### PR TITLE
Roles in user details

### PR DESCRIPTION
**Pull request by kmerz**
## Description
Display Roles on the Users Details Page

## Motivation and Context
The user wants to see the roles a user has.

### CODE REVIEW

**Comment by juju2112** [comment]
This is a test comment to see how Jenkins reacts. You can ignore it. 

Hi, Konrad!

File: graylog2-web-interface/src/components/users/PaginatedItem.jsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import styled, { css, type StyledComponent } from 'styled-components';
+
+import { type ThemeInterface } from 'theme';
+
+import type { DescriptiveItem } from './PaginatedItemOverview';
+
+type Props = { item: DescriptiveItem };
+
+const Container: StyledComponent<{ isAdmin: boolean }, ThemeInterface, HTMLSpanElement> = styled.span(({ isAdmin, theme }) => css`
+  display:block;
```
Comment (linuspahl) [inline graylog2-web-interface/src/components/users/PaginatedItem.jsx:12]: Lets add a space here.

Comment (linuspahl) [inline graylog2-web-interface/src/components/users/PaginatedItem.jsx:12]: Let's add a space here.

File: graylog2-web-interface/src/components/users/PaginatedItem.jsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+const Header = styled.h4`
+  padding-bottom: 5px;
+`;
+
+const PaginatedItem = ({ item: { name, description } }: Props) => (
+  <Container>
+    <Header>{name}</Header>
+    <span>{description}</span>
+    <hl />
```
Comment (linuspahl) [inline graylog2-web-interface/src/components/users/PaginatedItem.jsx:29]: Whats ist the purpose of `hl` here? I saw a warning in the console, that `hl` is not a supported html tag.

File: graylog2-web-interface/src/components/users/PaginatedItemOverview.jsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  onLoad: (PaginationInfo) => Promise<PaginatedListType>,
+};
+
+const PaginatedItemOverview = ({ onLoad }: Props) => {
+  const [items, setItems] = useState();
+  const [paginationInfo, setPaginationInfo] = useState({
+    count: 0,
+    total: 0,
+    page: 1,
+    perPage: 5,
```
Comment (linuspahl) [inline graylog2-web-interface/src/components/users/PaginatedItemOverview.jsx:46]: We could create a constant for the initial page size and use it here + for the `PaginatedList` -> `pagesSizes`

File: graylog2-web-interface/src/components/users/PaginatedItemOverview.jsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+  const _onSearch = (query) => {
+    const pageInfo = {
+      ...paginationInfo,
+      page: 1,
+      query,
+    };
+    onLoad(pageInfo).then(_setResponse);
+  };
+
```
Comment (linuspahl) [inline graylog2-web-interface/src/components/users/PaginatedItemOverview.jsx:77]: There are some warnings in the console for these methods:
`.then() only accepts functions but was passed: [object Undefined]`

File: graylog2-web-interface/src/components/users/PaginatedItemOverview.jsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  return (
+    <PaginatedList onChange={_onPageChange(paginationInfo.query)}
+                   pageSize={paginationInfo.perPage}
+                   totalItems={paginationInfo.total}
+                   pageSizes={[5, 10, 30]}
+                   activePage={paginationInfo.page}>
+      <SearchForm onSearch={_onSearch} />
+      <Container>
+        {items && items.toArray().map((item) => <PaginatedItem key={item.id} item={item} />) }
```
Comment (linuspahl) [inline graylog2-web-interface/src/components/users/PaginatedItemOverview.jsx:86]: We could display a message here, if there are no results for the current search.

File: graylog2-web-interface/src/components/users/UserDetails/RolesSection.jsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+const RolesSection = ({ user: { username } }: Props) => {
+  const [loading, setLoading] = useState(false);
+
+  const _onLoad = ({ page, perPage, query }: PaginationInfo): Promise<PaginatedListType> => {
+    setLoading(true);
+
+    // $FlowFixMe Role implements Descriptive Item!!!
+    return AuthzRolesActions.loadForUser(username, page, perPage, query)
+      .then(setLoading(false));
```
Comment (linuspahl) [inline graylog2-web-interface/src/components/users/UserDetails/RolesSection.jsx:23]: `() => setLoading(false)` 😉

Comment (linuspahl) [reply graylog2-web-interface/src/components/users/UserDetails/RolesSection.jsx:23]: (or something similar)

File: graylog2-web-interface/src/components/users/SectionComponent.jsx [hunk 7]
```diff
@@ -2,24 +2,30 @@
...
 const Headline = styled.h2`
   margin-bottom: 15px;
 `;
 
-const SectionComponent = ({ children, title }: Props) => (
+const SectionComponent = ({ children, title, showLoading = false }: Props) => (
   <Row className="content">
     <Col xs={12}>
-      <Headline>{title}</Headline>
+      <Headline>{title}</Headline>{showLoading && <Spinner />}
```
Comment (linuspahl) [inline graylog2-web-interface/src/components/users/SectionComponent.jsx:21]: I like it! but we should display the spinner, like in the `UsersOverview` currently `Loading...` is being displayed under the headline.

File: graylog2-web-interface/src/stores/roles/AuthzRolesStore.js [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+      const url = PaginationURL(ApiRoutes.AuthzRolesController.loadForUser(username).url, page, perPage, query);
+
+      const promise = fetch('GET', qualifyUrl(url))
+        .then(_responseToPaginatedList,
+          (error) => {
+            if (error.additional.status !== 404) {
+              UserNotification.error(`Loading roles for user ${username} failed with status: ${error}`,
+                'Could not load roles for user');
+            }
+          });
```
Comment (linuspahl) [inline graylog2-web-interface/src/stores/roles/AuthzRolesStore.js:60]: I prefer to trigger the user notification inside the component, so the store does not contain any presentation logic.  We should refactor this for our previous changes as well.

Comment (kmerz) [reply graylog2-web-interface/src/stores/roles/AuthzRolesStore.js:60]: Nice there was a bug in the code as well! :+1:

**Pull request merged**